### PR TITLE
Show error when offline mode is attempted without catalog.bin

### PR DESCRIPTION
**Pull request by twsouthwick**

### CODE REVIEW

File: src/ApiPort/App.config [hunk 1]
```diff
@@ -3,12 +3,13 @@
...
+    <trace autoflush="true" indentsize="4">
       <listeners>
-        <add name="myListener" type="System.Diagnostics.ConsoleTraceListener" />
+        <add name="logListener" type="System.Diagnostics.TextWriterTraceListener" initializeData="ApiPort.log" />
         <remove name="Default" />
       </listeners>
     </trace>
-  </system.diagnostics>
+  </system.diagnostics>-->
 </configuration>
```
Comment (mjrousos) [inline src/ApiPort/App.config]: Why the change to disable tracing?

File: src/Microsoft.Fx.Portability.Offline/Data.cs [hunk 2]
```diff
@@ -92,7 +92,7 @@ private static Stream OpenFileOrResource(string path)
 
                 if (stream == null)
                 {
-                    throw new FileNotFoundException("Could not find data file next to or embedded in assembly.", path);
+                    throw new PortabilityAnalyzerException($"Could not find data file '{path}' next to or embedded in assembly.");
```
Comment (mjrousos) [inline src/Microsoft.Fx.Portability.Offline/Data.cs]: Since this is a somewhat common error path, would it make sense to add a little more detail here about recommended next steps? Perhaps let the user know that the file should either be present when Microsoft.Fx.Portability.Offline is built or be placed next to the assembly?

File: src/ApiPort/App.config (hunk 1 above)
Comment (twsouthwick) [inline src/ApiPort/App.config]: Currently, it spews it out for everyone, when the messaging should let you know what the issue is. This change makes it so it logs it to a file and only when you explicitly enable it. I'm fine with enabling it by default, I just don't want it output to the console all the time.

Comment (mjrousos) [inline src/ApiPort/App.config]: Got it. Leaving it off by default seems fine.

File: src/Microsoft.Fx.Portability.Offline/Data.cs (hunk 2 above)
Comment (twsouthwick) [inline src/Microsoft.Fx.Portability.Offline/Data.cs]: Good call - I've updated the text. What do you think?

Comment (mjrousos) [inline src/Microsoft.Fx.Portability.Offline/Data.cs]: Looks good; thanks!

**Comment by mjrousos** [comment]
LGTM

**Comment by conniey** [comment]
LGTM
